### DISCUSSION TITLE
Parse Nested Lists

### DISCUSSION THREAD

**Discussion by vgowru** [ueberdosis/tiptap#1856] [General Questions] [answered]
Hi, 

When I pass nested list content to the editor with text in between the list items, the output is getting jumbled

```
<ul>
  <li>one</li>
  First
  <ul>
    <li>Two</li>
  </ul>
  Last
</ul>
```
(In realtime the text would be lodash template for loop)

The output is: 

<img width="321" alt="Screenshot 2021-09-07 at 3 11 31 PM" src="https://user-images.githubusercontent.com/56871294/132323015-223f9bf7-9e56-4fa8-b18f-42069be5c2a7.png">

I want the output to be 

<img width="321" alt="Screenshot 2021-09-07 at 3 13 58 PM" src="https://user-images.githubusercontent.com/56871294/132323398-7f98427a-897c-45ed-aa4c-41fc2d3b940e.png">

I have tried extending the bulletList extension and updating the content, but list-style won't show up when I try to add a new bullet list from the toolbar options.

How can I achieve the expected output?
Here is the code sandbox - https://codesandbox.io/s/tip-tap-editor-forked-owp7r

**Comment by cadars** [comment, marked as answer]
I think it's because your markup isn't valid HTML to start with, the nested list should be inside a `<li>` element of the list in which it is nested: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/ul

**Reply by NMVishnuVardhan** [reply]
@cadars Hi, I'm facing a similar issue. In my use case, I render the list items dynamically. 

```
      <ul>
       <% for(var comment of comments){ %>
         <li><%=comment.createdBy.firstName%> <%=comment.createdBy.lastName%>: <%=comment.content%></li>
       <% } %>
      </ul>
```

The editor changes this to below as bullet lists allow only listItem+ content.

```
     <ul>
       <li><% for(var comment of comments){ %></li>
         <li><%=comment.createdBy.firstName%> <%=comment.createdBy.lastName%>: <%=comment.content%></li>
       <li><% } %></li>
      </ul>
```
which results in

<img width="258" alt="Screenshot 2021-10-11 at 7 21 26 PM" src="https://user-images.githubusercontent.com/92315301/136801672-0342a9bf-bb14-4500-88b8-8936b61430c4.png">
 
Having the for loop above ul tag works but adds additional spaces as multiple `<ul>` s will be created.

<img width="258" alt="Screenshot 2021-10-11 at 7 26 09 PM" src="https://user-images.githubusercontent.com/92315301/136802412-cd000637-c6ca-4d81-97af-bb76ed1118cf.png">

I want the output to be single level bullet list
<img width="258" alt="Screenshot 2021-10-11 at 7 26 44 PM" src="https://user-images.githubusercontent.com/92315301/136802516-54223056-65f5-42ac-b7ba-49fea9b62223.png">

Any suggestions on how to implement loops inside ul without using CSS?

**Reply by philippkuehn** [reply]
Yeah it’s because this is not valid HTML.
